src/main/java/core/basesyntax/strategy/handler/BalanceHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Balance cannot be negative");
+        }
+        fruit.setQuantity(quantity);
+        Storage.fruits.put(fruit.getName(), fruit);
+    }
```

Comment:
The class `BalanceHandler` should not directly interact with `Storage`. Consider refactoring this to follow the Single Responsibility Principle. The `Storage` interaction should be encapsulated within a dedicated service.

Reply:
isn't it SPR?